### PR TITLE
Fix #400, better support for target opset

### LINKED ISSUE #400: Nightly build is broken.
This is due to the fact onnxruntime now checks an operator cannot be used as is for a opset higher than the last one declared. Many unit test needs to be updated. One example, the Following code fails with current Onnxruntime from master branch due to ``onnxruntime.capi.onnxruntime_pybind11_state.InvalidGraph: [ONNXRuntimeError] : 10 : INVALID_GRAPH : This is an invalid model. Error in Node: : No Op registered for Add with domain_version of 12``.

```
        sum_in = onnx.helper.make_tensor_value_info(
            'sum_in', onnx.TensorProto.FLOAT, [2])
        next = onnx.helper.make_tensor_value_info(
            'next', onnx.TensorProto.FLOAT, [2])
        sum_out = onnx.helper.make_tensor_value_info(
            'sum_out', onnx.TensorProto.FLOAT, [2])
        scan_out = onnx.helper.make_tensor_value_info(
            'scan_out', onnx.TensorProto.FLOAT, [2])
        add_node = onnx.helper.make_node(
            'Add',
            inputs=['sum_in', 'next'],
            outputs=['sum_out']
        )
        id_node = onnx.helper.make_node(
            'Identity',
            inputs=['sum_out'],
            outputs=['scan_out']
        )
        scan_body = onnx.helper.make_graph(
            [add_node, id_node],
            'scan_body',
            [sum_in, next],
            [sum_out, scan_out]
        )
        node = onnx.helper.make_node(
            'Scan',
            inputs=['initial', 'x'],
            outputs=['y', 'z'],
            num_scan_inputs=1,
            body=scan_body
        )

        initial = helper.make_tensor_value_info(
            'initial', TensorProto.FLOAT, [2, ])
        X = helper.make_tensor_value_info('x', TensorProto.FLOAT, [3, 2])
        Y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [2, ])
        Z = helper.make_tensor_value_info('z', TensorProto.FLOAT, [3, 2])

        graph_def = helper.make_graph(
            [node],
            'test-model',
            [initial, X],
            [Y, Z],
        )

        model_def = helper.make_model(graph_def, producer_name='onnx-example')

        initial = np.array([0, 0]).astype(np.float32).reshape((2,))
        x = np.array([1, 2, 3, 4, 5, 6]).astype(np.float32).reshape((3, 2))

        sess = InferenceSession(model_def.SerializeToString())
        res = sess.run(None, {'initial': initial, 'x': x})

        y = np.array([9, 12]).astype(np.float32).reshape((2,))
        z = np.array([1, 2, 4, 6, 9, 12]).astype(np.float32).reshape((3, 2))
        assert_almost_equal(y, res[0])
        assert_almost_equal(z, res[1])
```

### PR DESCRIPTION
**Pull request by xadupre**
* support target_opset as dictionary for multiple domain

### CODE REVIEW

File: .azure-pipelines/linux-CI-nightly.yml [hunk 1]
```diff
@@ -15,28 +15,22 @@ jobs:
     matrix:
       Py37-Onnx160-Sk0221:
         python.version: '3.7'
-        numpy.version: '==1.18.0'
+        numpy.version: '==1.18.1'
```
Comment (wenbingl) [inline .azure-pipelines/linux-CI-nightly.yml]: skl2onnx can only work with this latest numpy release?

File: skl2onnx/algebra/onnx_operator.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import numpy as np
 from scipy.sparse import coo_matrix
 from ..proto import TensorProto
-from ..common._topology import Variable, Scope
+from ..common._topology import Variable, Scope, _update_domain_version
```
Comment (wenbingl) [inline skl2onnx/algebra/onnx_operator.py]: No PEP warning about importing a private func?

File: .azure-pipelines/linux-CI-nightly.yml (hunk 1 above)
Comment (xadupre) [reply .azure-pipelines/linux-CI-nightly.yml]: No both. I usually choose the last one of series (1.17.5, 1.18.1).

File: tests/test_algebra_onnx_operators.py [hunk 3]
```diff
@@ -22,7 +22,11 @@
     helper, TensorProto, load_model,
     __version__ as onnx__version__
 )
-from test_utils import dump_data_and_model
+from test_utils import dump_data_and_model, TARGET_OPSET as TO11
+
+
+TARGET_OPSET = None
+TARGET_OPSET_11 = TO11
```
Comment (prabhat00155) [inline tests/test_algebra_onnx_operators.py]: Why are we naming it TARGET_OPSET_11? Wouldn't this need to be changed for 12, 13 ...?

File: tests/test_sklearn_one_vs_rest_classifier_converter.py [hunk 4]
```diff
@@ -264,21 +278,28 @@ def test_ovr_classification_int_binary_mlp(self):
...
         self.assertIsNotNone(model_onnx)
         dump_data_and_model(
-            X,
+            X[:5],
             model,
             model_onnx,
             basename="SklearnOVRRegressionFloat-Out0",
             allow_failure="StrictVersion(onnxruntime.__version__)"
-            "<= StrictVersion('0.2.1')",
+            "<= StrictVersion('1.1.0')",
```
Comment (prabhat00155) [inline tests/test_sklearn_one_vs_rest_classifier_converter.py]: Why this change?

File: skl2onnx/algebra/onnx_operator.py (hunk 2 above)
Comment (xadupre) [reply skl2onnx/algebra/onnx_operator.py]: I guess not, private Inside the module, users should not use it.

File: tests/test_sklearn_one_vs_rest_classifier_converter.py (hunk 4 above)
Comment (xadupre) [reply tests/test_sklearn_one_vs_rest_classifier_converter.py]: I changed again. The fact any change (random_state, reduce the test) I made did not seem to be efficient accross all versions.
